keyboards/royal_kludge/rk61/rgb/readme.md

```diff
@@ -0,0 +1,67 @@
+# RK61 RGB
+
+![RK61 RGB](https://i.pinimg.com/originals/01/e2/1f/01e21f1ba56526a0f2fe5047b6d25f11.png)
+
+Royal Kludge RK61 non-hotswap rgb edition, original PID 0x24F.
+
+Tested on the Royal Kludge RK61 non-hotswap RGB dual-mode edition, original PID 0x24F
+(RK61 firmware v1.03.000n, purchased after May 2020).
+
+- MCU: HFD2201KBA (rebrand SN32F248BF)
+- PCB: RK61-JX-RGB-HFD; REV: 01; 2019/08/13
+- Wired only, no bluetooth support.
+
+Keyboard Maintainer: [mr-july](https://github.com/mr-july)
+
+For the white-backlit non-rgb version of the RK61, check out [Diff-fusion's repo](https://github.com/Diff-fusion/qmk_firmware)
+
+
+Note: keymap differs slightly from stock firmware for FN combinations
+configuring RGB settings. Also adds ability to control volume and display
+brightness + additional FN keys.
+See [keymap.c](keymaps/default/keymap.c), and refer to the [list
+of QMK keycodes](https://beta.docs.qmk.fm/using-qmk/simple-keycodes/keycodes).
+
+-----------------
+
+
+Make example for this keyboard (after setting up your build environment):
+
+    make royal_kludge/rk61_rgb:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader:
+
+- Physical reset: short the two contacts to the left of the space bar switch with a wire/paperclip, accessible from the bottom of PCB.
+  ![HW bootloader](imgs/enter-bootloader.jpg)
+  After shorting, just connect the USB cable. The keyboard started in bootloader mode and is ready to be flashed.
+- Layout (after loading QMK): FN + CapsLock
+
```

Comment:
Maybe try adding Sonix flasher as an easy way to get into bootloader.

Reply:
As far as i know the Sonix flasher does not currently support the RK61. We need implement it first in Sonix flasher repo. (This is not a big deal, i hope). I'll look into it.

But we need testing to ensure, that the bootloader can be entered with the aid of Sonix flasher.

Reply:
Sorry, it was really a problem with ´keymap.h´ (this name seems to conflict with the system one), so I have renamed the file to ´layout.h´. Now your compile problem should be fixed.

Reply:
> As far as i know the Sonix flasher does not currently support the RK61. We need implement it first in Sonix flasher repo. (This is not a big deal, i hope). I'll look into it.
> 
> But we need testing to ensure, that the bootloader can be entered with the aid of Sonix flasher.

It kinda does, RK uses HFD, and by that, the keyboard gets put into bootloader. Currently, Sonix flasher supports putting into bootloader for HFD and eVision Sonix clone chips. By the way, if QMK gets installed in the keyboard, and you put it into bootloader, it will still get detected by Sonix flasher.

P.S.
Tested Sonix flasher on my end, it works for my little brother's keyboard. (at least for the bootloader part). I haven't tested flashing though, will update you once I've tested it.

Reply:
> Maybe try adding Sonix flasher as an easy way to get into bootloader.

I have just completed the documentation.

Reply:
> As far as i know the Sonix flasher does not currently support the RK61.

Actually it is supported. But it is detected as "Apple Keyboard / Keychron / Flashquark Horizon Z" if the original firmware is installed. If the QMK is installed, then it will not be detected.

> We need implement it first in Sonix flasher repo. (This is not a big deal, i hope). I'll look into it.

The following patch need to be applied to Sonix flasher to add support for RK61.

```diff
diff --git a/src/main/python/main.py b/src/main/python/main.py
index 586a7e3..288e137 100644
--- a/src/main/python/main.py
+++ b/src/main/python/main.py
@@ -41,6 +41,7 @@ DEVICE_DESC = {
     (0x0c45, 0x766b): "Kemove",
     (0x0c45, 0x7698): "Womier",
     (0x320f, 0x5013): "Akko",
+    (0x0c45, 0x024f): "Royal Kludge RK61 RGB",
     (0x0c45, 0x5004): "Redragon",
     (0x0c45, 0x5104): "Redragon",
     (0x0C45, 0x8513): "Sharkoon",
```

Should I create a PR for this?

Actually I'm not sure which `VENDOR_ID` and `PRODUCT_ID` need to be used for QMK. Are any rules or guidelines existing?

Reply:
> > As far as i know the Sonix flasher does not currently support the RK61.
> 
> Actually it is supported. But it is detected as "Apple Keyboard / Keychron / Flashquark Horizon Z" if the original firmware is installed. If the QMK is installed, then it will not be detected.
> 
> > We need implement it first in Sonix flasher repo. (This is not a big deal, i hope). I'll look into it.
> 
> The following patch need to be applied to Sonix flasher to add support for RK61.
> 
> ```diff
> diff --git a/src/main/python/main.py b/src/main/python/main.py
> index 586a7e3..288e137 100644
> --- a/src/main/python/main.py
> +++ b/src/main/python/main.py
> @@ -41,6 +41,7 @@ DEVICE_DESC = {
>      (0x0c45, 0x766b): "Kemove",
>      (0x0c45, 0x7698): "Womier",
>      (0x320f, 0x5013): "Akko",
> +    (0x0c45, 0x024f): "Royal Kludge RK61 RGB",
>      (0x0c45, 0x5004): "Redragon",
>      (0x0c45, 0x5104): "Redragon",
>      (0x0C45, 0x8513): "Sharkoon",
> ```
> 
> Should I create a PR for this?
> 
> Actually I'm not sure which `VENDOR_ID` and `PRODUCT_ID` need to be used for QMK. Are there any rules or guidelines?

You don't have to make a patch for Sonix Flasher though you can if you want to. Anyway, regarding the `VID` and `PID`, only the Keychron has guidelines around that. You can copy the stock firmware's `PID` and `VID` provided that they don't mimic the Apple keyboard.

Reply:
> Anyway, regarding the `VID` and `PID`, only the Keychron has guidelines around that.

Could you provide me an info link, please?

> You can copy the stock firmware's `PID` and `VID`, provided that they don't mimic the Apple keyboard.

But for RK61 RGB, that is just the case. Original firmware masquerades as an Apple keyboard (05ac:024f). So, it is probably not an option just to use the original `PID` and `VID`. So, we need change it. But the values currently used (0c45:024f) are unknown for the Sonix Flasher. So, after first flash (from original firmware to self compiled QMK), the second flash will not be possible with the Sonix Flasher, because the keyboard (0c45:024f) will not be identified. It is necessary to manually switch the keyboard into bootloader first, then it will be recognized (0c45:7040) and can be flashed. Is it so for other keyboards too?

Reply:
> Could you provide me an info link, please?
Refer to #152 

> But for RK61 RGB, that is just the case. Original firmware masquerades as an Apple keyboard (05ac:024f). So, it is probably not an option just to use the original PID and VID. So, we need change it. But the values currently used (0c45:024f) are unknown for the Sonix Flasher. So, after first flash (from original firmware to self compiled QMK), the second flash will not be possible with the Sonix Flasher, because the keyboard (0c45:024f) will not be identified. It is necessary to manually switch the keyboard into bootloader first, then it will be recognized (0c45:7040) and can be flashed. Is it so for other keyboards too?
If RK61 RGB doesn't mimic Apple Keyboard's `PID` and `VID`, then you don't have to change it. To address your next question too, Sonix Flasher cannot put the keyboard anyway to bootloader mode when you have already flashed QMK, so that's fine.